components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -142,30 +147,56 @@ - (UIColor *)titleColorForState:(UIControlState)state {
 
 #pragma mark - UIView
 
-// TODO(#7645): Temporary layout until
-// https://github.com/material-components/material-components-ios/issues/7645 lands
 - (void)layoutSubviews {
   [super layoutSubviews];
 
-  if (self.items.count == 0) {
-    return;
+  CGFloat availableWidth = CGRectGetWidth(self.bounds);
+  CGFloat maxWidth = 0;
+  for (UIView *itemView in self.containerView.arrangedSubviews) {
+    CGSize contentSize = itemView.intrinsicContentSize;
+    if (contentSize.width > maxWidth) {
+      maxWidth = contentSize.width;
+    }
   }
-  CGSize boundsSize = CGSizeMake(CGRectGetWidth(self.bounds), CGRectGetHeight(self.bounds));
-  CGSize itemSize = CGSizeMake(boundsSize.width / self.items.count, boundsSize.height);
+  CGFloat requiredWidth = maxWidth * self.items.count;
+  BOOL canBeJustified = availableWidth >= requiredWidth;
+  self.containerView.distribution = canBeJustified ? UIStackViewDistributionFillEqually
+                                                   : UIStackViewDistributionFillProportionally;
+}
 
-  CGPoint itemOrigin = CGPointMake(CGRectGetMinX(self.bounds), CGRectGetMinY(self.bounds));
-  for (UIView *itemView in self.itemViews) {
-    itemView.frame = CGRectMake(itemOrigin.x, itemOrigin.y, itemSize.width, itemSize.height);
-    itemOrigin = CGPointMake(itemOrigin.x + itemSize.width, itemOrigin.y);
+- (void)updateConstraints {
+  if (self.containerViewConstraintsActive) {
+    [super updateConstraints];
+    return;
   }
+
+  [_containerView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor].active = YES;
+  [_containerView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor].active = YES;
+  [_containerView.widthAnchor constraintGreaterThanOrEqualToAnchor:self.widthAnchor].active = YES;
+  [_containerView.topAnchor constraintEqualToAnchor:self.topAnchor].active = YES;
+  [_containerView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor].active = YES;
+  self.containerViewConstraintsActive = YES;
+
+  // Must always be called last according to the documentation.
+  [super updateConstraints];
 }
 
 - (CGSize)intrinsicContentSize {
```

Comment:
Please add unit tests to cover verify the behavior/values of `intrinsicContentSize`.

Reply:
Is it worth to import MDCTabBarItemView into this test to verify the exact size the intrinsicContentSize should return?

Reply:
No, I don't think the TabBarView should be dependent on its contents.  I'd prefer the tests cover the more general behavior of `intrinsicContentSize`. For example, ensuring that it returns the justified/scrollable width consistently (depending on what you determine is more useful). That the height grows if one of its items has a very large image or long text.  It's the relationship/behavior that's important and not the exact values.

Let me add one or two as a commit and we can talk about them.  I think that will be faster and clearer than my trying to explain in a comment. ;-)